sql-common/my_time.c

```diff
@@ -336,12 +339,25 @@ str_to_datetime(const char *str, uint length, MYSQL_TIME *l_time,
     pos++;
     skip_digits(&pos, end); // ignore the return value
   }
+  if (pos < end && *pos == '-' && digits >= 12) /* YYYYYMMDDHHMMSShhmmss.uuuuuu-HH:MM is supported too */
+  {
+    pos+=6; // Skip 6 places as defined in RFC3339 reserved for local offset
+  }
+  else if (pos < end && *pos == '+' && digits >= 12) /* YYYYYMMDDHHMMSShhmmss.uuuuuu+HH:MM is supported too */
+  {
+    pos+=6; // Skip 6 places as defined in RFC3339 reserved for local offset
```

Comment:
I am working on adding support for parsing and changing the parsed datetime to system configured timezone. 

Parsing the time changes the current behavior of just truncating the datetime object. Would this be accepted into the 10.1 branch? Or would it be better to push that feature to 10.2? I have a need in the 10.1 branch for atleast this patch's level of support for rfc3339 in order to get partition pruning working.